src/main/java/Duke.java

```diff
@@ -1,10 +1,46 @@
+import java.util.Scanner;
+
+/**
+ * Defines the Duke class for the end-user to manage tasks.
+ *
+ * @author Kai Chao
+ * @version 1.0
+ * @since 26-08-2020
+ */
 public class Duke {
+    private Storage storage;
+    private TaskList tasks;
+    private Ui ui;
+    private Command command;
+
+    Duke(String dirPath, String filePath) {
+        this.ui = new Ui();
+        this.storage = new Storage(dirPath, filePath);
+        this.tasks = new TaskList(storage.loadData());
+        this.command = new Command();
+    }
+
     public static void main(String[] args) {
-        String logo = " ____        _        \n"
-                + "|  _ \\ _   _| | _____ \n"
-                + "| | | | | | | |/ / _ \\\n"
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+        new Duke("./data", "./data/tasks.txt").run();
+    }
+
+    /** Runs the Duke program. */
+    public void run() {
+        command.welcomeCommand();
+        Scanner scanner = new Scanner(System.in);
+        boolean terminated = false;
```

Comment:
Perhaps it may be more intuitive to rename the variable "terminated" to "isTerminated"?

Reply:
Thanks, missed that one out!